kaldb/src/main/java/com/slack/kaldb/writer/kafka/KaldbKafkaConsumer.java

```diff
@@ -38,57 +41,81 @@ public class KaldbKafkaConsumer {
-      String enableKafkaAutoCommit,
-      String kafkaAutoCommitInterval,
-      String kafkaSessionTimeout,
-      Properties overrideProps) {
+  @VisibleForTesting
+  public static Properties makeKafkaConsumerProps(KaldbConfigs.KafkaConfig kafkaConfig) {
+
+    String kafkaBootStrapServers = kafkaConfig.getKafkaBootStrapServers();
+    String kafkaClientGroup = kafkaConfig.getKafkaClientGroup();
+    String enableKafkaAutoCommit = kafkaConfig.getEnableKafkaAutoCommit();
+    String kafkaAutoCommitInterval = kafkaConfig.getKafkaAutoCommitInterval();
+    String kafkaSessionTimeout = kafkaConfig.getKafkaSessionTimeout();
 
     Properties props = new Properties();
     props.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, kafkaBootStrapServers);
     props.put(ConsumerConfig.GROUP_ID_CONFIG, kafkaClientGroup);
     // TODO: Consider committing manual consumer offset?
     props.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, enableKafkaAutoCommit);
     props.put(ConsumerConfig.AUTO_COMMIT_INTERVAL_MS_CONFIG, kafkaAutoCommitInterval);
+    // TODO: Does the session timeout matter in assign?
+    props.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, kafkaSessionTimeout);
+
     props.put(
         ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG,
         "org.apache.kafka.common.serialization.StringDeserializer");
-    // TODO: Using ByteArrayDeserializer since it's most primitive and performant. Replace it if
-    // not.
     props.put(
         ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG,
         "org.apache.kafka.common.serialization.ByteArrayDeserializer");
-    // TODO: Does the session timeout matter in assign?
-    props.put(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, kafkaSessionTimeout);
+
     // we rely on the fail-fast behavior of 'auto.offset.reset = none' to handle scenarios
     // with recovery tasks where the offsets are no longer available in Kafka
     props.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "none");
 
-    if (overrideProps != null && !overrideProps.isEmpty()) {
-      props.putAll(overrideProps);
+    // don't override the properties that we have already set explicitly using named properties
+    for (Map.Entry<String, String> additionalProp :
+        kafkaConfig.getAdditionalPropsMap().entrySet()) {
+      maybeOverride(
+          props,
+          additionalProp.getKey(),
+          additionalProp.getValue(),
+          OVERRIDABLE_CONFIGS.contains(additionalProp.getKey()));
     }
-
     return props;
   }
 
+  @VisibleForTesting
+  public static boolean maybeOverride(
+      Properties props, String key, String value, boolean override) {
+    boolean overridden = false;
+    String userValue = props.getProperty(key);
+    if (userValue != null) {
+      if (override) {
+        LOG.warn(
+            String.format(
+                "Property %s is provided but will be overridden from %s to %s",
+                key, userValue, value));
+        props.setProperty(key, value);
+        overridden = true;
+      }
```

Comment:
Maybe also warn in the case that the user specifies a Kafka property in the `addtionalPropsMap` that cannot be overridden?